src/java/org/apache/cassandra/transport/Dispatcher.java

```diff
@@ -79,17 +80,67 @@ public Dispatcher(boolean useLegacyFlusher)
 
     public void dispatch(Channel channel, Message.Request request, FlushItemConverter forFlusher, Overload backpressure)
     {
-        requestExecutor.submit(() -> processRequest(channel, request, forFlusher, backpressure));
+        requestExecutor.submit(new RequestProcessor(channel, request, forFlusher, backpressure));
```

Comment:
The fun thing for testing is that in-JVM dtests don't dispatch normally against the NTR stage pool. It might end up being bette to use `CQLTester` for this...